code401challenges/src/main/java/tree/BinarySearchTree.java

```diff
@@ -42,6 +47,36 @@ public boolean contains(int data)
         return contains(root_Node, data);
     }
 
+    private ArrayList<Integer> breadthFirst(Node node, ArrayList<Integer> list)
+    {
+        if(node == null)
+            return null;
+
+
+        Queue<Node> node_Queue = new Queue<>();
+        node_Queue.enqueue(node);
+
+        while (node_Queue.peek() != null)
```

Comment:
Can simplify to `while(!node_Queue.isEmpty())`... this expresses what we want: "while the queue is not empty"